zebra-chain/src/transaction.rs

```diff
@@ -112,6 +105,8 @@ pub enum Transaction {
         inputs: Vec<transparent::Input>,
         /// The transparent outputs from the transaction.
         outputs: Vec<transparent::Output>,
+        /// The sapling shielded data for this transaction, if any.
+        sapling_shielded_data: Option<sapling::ShieldedData<sapling::SharedAnchor>>,
```

Comment:
I would like to review this change in two separate PRs:
1. Changes to sapling_shielded_data in V4
2. Adding sapling_shielded_data to V5

That way, we can get the `V4` changes merged first.